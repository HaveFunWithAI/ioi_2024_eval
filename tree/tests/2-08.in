2000
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
965365 96970 222027 111278 410562 993053 148350 383847 401925 993437 370403 459350 446315 881408 443737 443248 263766 383518 96063 361613 304842 812761 833264 778401 496337 653315 412010 78278 651407 118755 121924 798168 917409 832919 166624 516302 9967 308545 237675 493199 998119 475859 726963 787154 747180 257902 930343 806219 20166 167252 440452 267710 618008 261961 239420 986217 176973 322096 923075 345367 347394 107580 43987 445985 503811 959223 743329 851140 351910 464778 396223 353567 473194 226841 513361 441538 550381 768626 701208 528291 956482 80295 207828 736168 920815 66475 483762 81842 969027 654498 980559 471446 621524 747819 539961 762521 191163 508805 84541 157904 519087 187470 146805 413899 841166 737501 911413 143911 218306 590303 288976 858483 155061 121765 838329 64196 511768 964904 475359 533056 753935 150019 871147 718005 438024 570715 452029 430811 315943 202960 410166 14216 422295 132459 382420 158596 134002 718316 965901 380569 88689 130132 426328 319831 969008 335611 162282 616588 935616 74204 23830 20931 921485 207460 399925 506184 248923 3783 591086 902104 994342 690604 84562 345314 821302 826267 256078 845642 823065 615522 231294 305352 557244 892129 610537 355640 802037 657946 686399 310083 359575 440599 327949 265191 202267 127051 69684 221212 53801 204902 679008 338571 967034 84963 689091 995647 977651 237084 387535 522273 175785 185926 873618 960963 92502 600977 233151 459202 73885 54018 559136 768618 704176 612237 51450 249447 309630 688071 360783 119643 414216 378567 215766 131044 603618 361411 147229 564854 262243 226858 170974 246266 337894 628935 849995 936825 449743 519513 334572 37682 555108 672587 911750 862634 328668 821870 40549 230901 238620 520299 960768 512487 602266 934946 879551 305660 917686 502120 796993 590088 643651 656173 421793 654977 531063 917684 962703 728423 949007 649057 586211 755550 219755 401659 514236 990103 860689 741244 514740 308988 614487 642556 689007 299179 847676 289448 450033 479919 159939 664347 187432 136318 504537 720077 659408 659105 935974 794041 59031 262257 26675 176450 306670 31001 303471 209079 224980 782776 59784 831672 658527 448780 287090 558244 953049 948447 185472 692967 966714 416123 810523 496756 677524 976679 182824 488805 33072 896707 140084 117710 756641 666311 758689 722978 615834 893587 663526 339167 652262 985284 912803 57433 175022 858141 781462 160518 43412 368599 308998 317593 190549 934140 229167 278088 458020 518334 585884 199812 619593 274165 306390 316377 708393 492599 993455 878907 215722 774625 917686 531642 297607 940507 245870 552847 325871 924928 204043 858422 471261 459524 34556 209911 972981 77525 78070 457743 818566 112196 507299 622442 208063 310817 147149 991363 852452 640676 980221 589743 533367 59079 397683 945327 459532 470994 327404 638920 300012 199345 261099 528953 685835 562544 426210 870619 21324 360434 969868 190715 390868 863481 14218 401677 88789 324227 600689 98079 892117 95232 3424 856369 890407 397504 4192 180812 972713 615149 386188 291045 19541 822889 727647 949192 757693 633984 377539 89552 243241 973755 793260 368601 910653 402185 331151 728832 296785 119986 850604 602914 700925 749315 366239 734135 557362 837250 89145 561243 553358 315278 248967 473714 794993 752623 554705 611643 94716 996548 273752 30969 375625 728953 728174 534244 882296 782664 234267 315839 191566 828430 101310 65110 180481 811935 440969 22178 141602 250985 192336 401833 670579 621285 134057 308741 821727 186412 447989 174621 636962 201169 886095 297188 74239 846119 976853 903792 235158 260962 908899 671222 252658 630890 893272 108393 490556 959415 68024 761324 881493 751217 767757 651952 596741 266108 206422 716560 101217 91345 501608 396112 555699 777244 103826 916519 404796 151818 153196 226077 977688 986310 827222 754270 993804 11326 199876 96798 232434 215899 365939 194635 869670 102143 664313 34073 903358 902890 265317 859547 595176 214700 435986 269377 701594 108461 189957 27056 673549 775748 304725 642565 906183 151872 608541 558895 155470 212684 713919 82102 112116 234398 859486 5133 566429 992103 62734 511059 628352 667348 246490 340328 749337 974752 253756 14976 535805 944057 52611 694638 733744 249147 22572 308998 419182 274253 852145 398810 29127 584261 220814 227892 696231 226539 162374 841492 304552 11731 283851 897466 777306 686821 174515 369146 667735 710936 731007 748411 550042 185084 815021 598777 142046 907445 743430 601645 655911 889856 592877 363726 734680 896511 46560 186428 876877 816174 341383 61138 624656 741243 731640 64766 769100 439716 436223 88020 119985 498046 781776 671632 963871 338660 890424 921285 599399 228565 197322 158088 666155 338972 160177 449605 410819 16794 645550 736063 464954 174508 761112 188795 667873 327241 589315 512906 904275 323687 415182 523811 966905 10149 526893 469694 503866 513181 771294 889113 312853 946068 72760 212527 8847 960549 231392 544056 730522 653895 14496 955888 805769 169713 132856 899703 738269 895582 85862 419350 865247 556121 335261 240263 51126 556522 688382 552432 76115 69314 554553 57092 326583 196987 707576 473490 412989 724794 373889 362416 113065 977191 181433 981652 383897 706162 26202 142402 95937 865291 390568 134884 481061 645837 217585 723009 794112 533332 206860 929914 481566 960301 402869 417500 182203 812901 886577 208782 492398 966698 330000 77407 866555 124983 261332 894352 214797 417653 154608 870210 509935 874738 785262 414674 128379 140089 668758 431607 362222 184280 978231 17863 122544 79618 512814 726888 959997 107612 321299 691644 547964 805154 877905 208417 266550 713281 19895 369909 583831 237870 439084 347944 634075 898662 147809 145454 358519 297924 761619 196111 311031 920733 499133 893618 570526 4290 943444 167665 573127 941144 916998 308363 380431 641963 281480 441152 264319 26849 175493 80895 70829 810429 118822 417876 828921 393866 535199 72350 458484 827525 17550 530613 302178 671959 264170 221060 149563 883836 372568 537532 822645 559748 429226 539841 343889 100214 240537 211461 207704 433313 1422 443148 759331 500226 250025 529312 439192 415468 350678 748765 369533 641370 236814 618944 999133 626561 853061 342250 408206 873736 860977 440158 169308 659779 127932 544045 590118 583648 704069 879636 345829 528395 510904 243706 17738 60735 53545 934813 3534 332946 689873 922072 927102 751073 501562 146393 585387 531246 631547 778788 167294 108500 816965 191795 857470 139183 355130 790572 148430 587410 295886 789314 568814 144666 569000 248240 447517 632241 157143 775417 806807 480124 901567 478444 175719 966917 178173 553463 388546 238109 579335 282012 818851 651168 439709 998518 358517 870052 259620 609880 950479 728432 132294 255886 450628 163402 369960 476816 45151 644030 554663 911437 675750 507596 15916 916795 770517 491745 20242 641058 190179 764435 427157 362913 665520 783136 381023 515625 201223 89528 596480 974260 538223 419046 745527 562349 813125 757224 569493 50301 716728 496488 955294 661799 852261 674508 164768 550204 545697 845683 188107 872062 564916 288054 736596 580712 788828 166992 375188 716511 14213 426467 267754 454098 212503 761101 89223 715755 528845 721665 133847 415804 135512 600532 223109 849903 265316 623416 618895 634480 578030 812560 768053 797021 955224 135312 259777 394158 811921 472191 874298 317488 54757 658362 928435 55065 67822 37067 241687 141288 551409 365598 390950 160566 685698 577066 764877 597583 363932 144230 227678 617234 954856 983808 17662 559141 348527 708534 418945 284214 585564 934532 430590 154268 755860 824437 943998 426856 277167 484439 786613 193059 331483 601536 438226 897273 102438 903553 488690 92829 523266 160758 956799 91878 262566 711731 8621 434817 961964 709597 41704 479780 84199 230212 267578 674528 883039 362512 392827 879019 964024 267123 401083 580820 374936 151476 736069 903235 29604 410051 6719 281276 144363 681315 360434 920454 263168 160761 577142 151119 101109 719430 247588 401 593031 431082 140360 607063 852189 764973 621830 884465 801717 433338 705001 217132 87275 364431 357917 587608 52223 783103 566230 477466 443277 911388 443739 824320 402381 871762 957267 303846 929861 299710 655853 766688 702390 55111 795685 569072 779077 168745 529454 432825 901356 727705 19640 232478 868414 406836 135864 502686 451534 128172 962452 344902 351365 632588 234282 742411 886295 140259 807376 967692 12705 125817 378695 467127 432938 493091 417534 366965 793620 49936 422803 405362 852806 351917 647968 916930 897889 917485 532718 959557 173769 528972 975041 487544 213692 946620 800295 870616 782685 48077 255752 904983 982193 434677 118137 551277 611846 101427 731353 232050 446668 3766 648728 329362 887389 603956 942978 646777 602896 347780 560927 928127 212727 660211 307148 213201 214158 913446 990886 315724 564253 978772 574712 645249 952266 952339 122605 262228 511803 530285 929772 580921 684575 221736 762343 340629 481471 806229 342013 707376 643174 881041 640432 282657 879827 219268 295232 87605 314499 942879 48967 384394 765186 139544 878109 832607 808654 296007 34734 297977 601975 574732 992319 167989 152394 895851 446288 414995 432964 511432 58204 408736 2485 859781 602454 918668 19899 224146 553061 993567 150572 896719 719367 657038 970111 305734 861660 132008 74600 732328 308897 217123 937387 160313 717455 523108 690366 800159 768835 698402 903030 790147 277262 557489 669307 82534 223611 561675 854705 122899 44325 604049 265216 753682 272142 879896 809056 972667 205128 32086 349725 399614 561439 546670 372064 487695 524155 94774 84030 36113 547572 163399 576473 638929 352531 815582 702194 538817 378228 549335 365337 516217 359296 726170 841888 387901 783443 617280 455028 703541 359969 493525 651758 645419 355691 161802 982467 394955 800837 541301 447010 164952 284705 697894 224101 826453 410246 914391 523539 395190 645363 383693 325259 372291 884932 697144 314484 460810 165095 773234 657351 384876 548438 420152 227232 488014 676983 420217 506250 410957 81083 106603 996194 950531 853515 381013 833257 834860 644529 725639 832363 21820 520196 276314 634715 665411 742079 651466 229272 384356 301088 816883 361236 199570 156574 559267 413227 609616 888971 145594 868096 527443 839332 376512 476813 841209 291847 577132 740969 701207 585652 200655 517812 817574 65813 166235 644222 312646 295138 464253 450346 502269 40141 425822 300298 691302 15629 165173 778345 912417 991973 755221 943770 860747 601277 550175 21850 758260 157660 856064 537305 85682 50097 155532 889482 344679 228825 285522 262658 517846 861499 48700 670658 55098 119774 917101 760374 518525 580266 504300 85301 388535 799674 489527 378715 127109 239957 981071 553328 455262 232227 6645 618202 535666 752357 846558 642041 995847 493942 376952 803479 509244 684383 911 587617 348932 950879 123899 725441 992601 691322 95653 686241 740524 8820 381754 250895 602757 617044 606009 752822 778767 187697 922980 19478 28663 802865 155930 102856 976293 545110 454698 647008 848731 232286 58060 52368 32379 713971 106204 437458 288248 904260 905814 312183 503094 515269 204479 283833 756848 236232 238471 464560 355601 607566 8823 588484 321923 994076 514750 753789 341073 80328 341382 481890 939560 466730 41570 758732 829896 670 283006 575381 696351 271529 75724 481465 674448 999078 185611 233937 423740 460825 214594 126714 603446 656804 496197 593671 572901 17471 186417 213859 852991 988691 260927 947180 82200 531231 886418 882903 831720 451488 925305 101123 168349 308910 299305 367730 526655 903297 192578 846095 466888 593481 451057 634480 942429 868745 561225 302654 633545 640646 994859 476711 605272 383639 9847 490021 928051 967331 266007 453126 829851 312523 709535 146543 426304 335341 109849 865348 642765 314596 604777 276400 982030 903074 725397 637396 299915 107287 417854 209987 151651 296868 458790 354793 370649 631761 146173 637528 457553 763463 456207 681359 54624 807726 17178 258438 320781 100791 589906 400643 868458 441114 267950 462665 831240 610003 112302 960629 989987 663069 119657 675871 80805 329100 583740 385956 149692 283865 199508 427816 778478 711113 905639 443923 128712 148445 150740 819094 838430 166883 819770 612252 790915 10040 161681 52646 669506 790746 18959 630365 64425 719632 438435 733307 12010 958004 928260 635191 166387 408200 99504 413832 129236 430898 709640 924761 885982 932972 601971 116626 571339 22004 687585 197890 98955 237725 632395 636994 565190 958600 29652 279799 325549 914553 66835 270769 198298 362865 152976 196453 660740 293936 696985 335254 227023 833847 346848 333819 536997 796821 495282 337864 757983 173036 999571 685873 304651 423697 280400 923084 446767 257172 324435 83067 384722 945715 32429 459060 736685 232120 534142 979841 786610 912831 105353 734666 401902 553764 716846 656439 612872 837651 219960 22452 744102 199025 839664 743701 652704 544662 355964 241362 10916 149416 52150 967534 192555 175650 302374 938821 88199 636930 228244 638949 54956 649029 308663 227265 944090 854637 499739 175934 604346 571168 631467 805424 417519 191153 936246 663427 270972 409812 228449 473916 240859 14848 506466 997511 797839 830572 321300 717892 680798 42933 30115 120078 249796 158678 955305 650717 453836 686254 305295 278763 991435 315327 367478 726599 729836 168200 112254 239466 665933 780382 176341 25408 185104 717479 39177 825034 594148 969627 31794 290117 721748 454050 748313 691644 882849 103438 497332 362159 779265 444338 344193 646124 39826 209091 353726 453810 452322 439010 871793 513495 554439 622076 228063 705352 448511 953879 659770 445151 264322 919192 779829 754010 590119 886049 924667 671164 450195 513871 868516 519994 36632 56862 237960 502411 434682 135606 788269 806423 479758 845993 170955 713882 898274 752291 923634 376889 245406 426350 601282 918638 128289 794498 669237 849725 267211 369639 977216 309901 630602 944328 241432 236385 907468 330687 731168 375842 867426
10
826646 837122
888974 892318
928872 962375
903789 981671
907500 969508
886286 965824
937394 966314
883251 916175
904606 965669
778969 824385
