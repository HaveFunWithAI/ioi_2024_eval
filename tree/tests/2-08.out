2405489897539184
2586868257938796
2702940400172773
2629907237390536
2640721392702080
2578972752495714
2727743433629036
2570186048325034
2632300904480169
2266718396003546
